Flaky `TestConvertOTelHistograms`

```
--- FAIL: TestConvertOTelHistograms (0.00s)
      otel_test.go:372: 
          	Error Trace:	/__w/mimir/mimir/pkg/distributor/otel_test.go:372
          	Error:      	"[]" should have 1 item(s), but has 0
          	Test:       	TestConvertOTelHistograms
  2025/04/24 18:03:31 label __name__ is overwritten. Check if Prometheus reserved labels are used.
  2025/04/24 18:03:31 label __name__ is overwritten. Check if Prometheus reserved labels are used.
  2025/04/24 18:03:31 label __name__ is overwritten. Check if Prometheus reserved labels are used.
  2025/04/24 18:03:31 label __name__ is overwritten. Check if Prometheus reserved labels are used.
  2025/04/24 18:03:31 label __name__ is overwritten. Check if Prometheus reserved labels are used.
  level=info msg="server listening on addresses" http=127.0.0.1:38721 grpc=127.0.0.1:38899
  level=warn method=/httpgrpc.HTTP/Handle duration=330.544µs request="&HTTPRequest{Method:POST,Url:/otlp,Headers:[]*Header{&Header{Key:Content-Type,Values:[application/x-protobuf],},},Body:[105 110 118 97 108 105 100],}" msg=gRPC err="rpc error: code = Code(400) desc = unexpected EOF"
  level=warn method=/httpgrpc.HTTP/Handle duration=601.328µs request="&HTTPRequest{Method:POST,Url:/otlp,Headers:[]*Header{&Header{Key:Content-Type,Values:[application/json],},},Body:[10 246 22 10 211 2 10 29 10 17 99 111 110 116 97 105 110 101 114 46 114 117 110 116 105 109 101 18 8 10 6 100 111 99 107 101 114 10 39 10 18 99 111 110 116 97 105 110 101 114 46 104],}" msg=gRPC err="rpc error: code = Code(400) desc = ReadObjectCB: expect { or n, but found \ufffd, error found in #2 byte of ...|\n\ufffd\u0016\n\ufffd\u0002\n\u001d\n\u0011co|..., bigger context ...|\n\ufffd\u0016\n\ufffd\u0002\n\u001d\n\u0011container.runtime\u0012\u0008\n\u0006docker\n'\n\u0012container.h|..."
  level=warn method=/httpgrpc.HTTP/Handle duration=542.298µs request="&HTTPRequest{Method:POST,Url:/otlp,Headers:[]*Header{&Header{Key:Content-Type,Values:[application/json],},},Body:[123 34 114 101 115 111 117 114 99 101 77 101 116 114 105 99 115 34 58 32 91 123 34 115 99 111 112 101 77 101 116 114 105 99 115 34 58 32 91 123 34 109 101 116 114 105 99 115 34 58 32 91 123 34 110 97 109 101 34 58 32 34 114 101 112 111 114 116 95 115 101 114 118 101 114 95 101 114 114 111 114 34 44 32 34 103 97 117 103 101 34 58 32 123 34 100 97 116 97 80 111 105 110 116 115 34 58 32 91 123 34 116 105 109 101 85 110 105 120 78 97 110 111 34 58 32 34 49 54 55 57 57 49 50 52 54 51 51 52 48 48 48 48 48 48 48 34 44 32 34 97 115 68 111 117 98 108 101 34 58 32 49 48 46 54 54 125 93 125 125 93 125 93 125 93 125],}" msg=gRPC err="rpc error: code = Code(503) desc = some random push error"
  level=warn method=/httpgrpc.HTTP/Handle duration=261.997µs request="&HTTPRequest{Method:POST,Url:/otlp,Headers:[]*Header{},Body:[104 101 108 108 111],}" msg=gRPC err="rpc error: code = Code(415) desc = unsupported content type: , supported: [application/json, application/x-protobuf]"
  level=warn method=/httpgrpc.HTTP/Handle duration=356.172µs request="&HTTPRequest{Method:POST,Url:/otlp,Headers:[]*Header{&Header{Key:Content-Type,Values:[application/json],},},Body:[105 110 118 97 108 105 100],}" msg=gRPC err="rpc error: code = Code(400) desc = ReadObjectCB: expect { or n, but found i, error found in #1 byte of ...|invalid|..., bigger context ...|invalid|..."
  level=info msg="=== Handler.Stop()'d ==="
```

--- a/pkg/distributor/otel_test.go
+++ b/pkg/distributor/otel_test.go
@@ -368,8 +368,22 @@ func TestConvertOTelHistograms(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, 0, dropped)
 		if convertHistogramsToNHCB {
-			require.Len(t, mimirTS, 2)
-			require.Len(t, mimirTS[0].Histograms, 1)
+			ts := make([]mimirpb.PreallocTimeseries, 0, len(mimirTS))
+			for i := range mimirTS {
+				var metricName string
+				for _, lbl := range mimirTS[i].Labels {
+					if lbl.Name == labels.MetricName {
+						metricName = lbl.Value
+						break
+					}
+				}
+				if metricName == "target_info" {
+					continue
+				}
+				ts = append(ts, mimirTS[i])
+			}
+			require.Len(t, ts, 1)
+			require.Len(t, ts[0].Histograms, 1)
 		} else {
 			require.Len(t, mimirTS, 6)
 			require.Len(t, mimirTS[0].Histograms, 0)

--- a/pkg/distributor/otel_test.go
+++ b/pkg/distributor/otel_test.go
@@ -369,6 +369,7 @@ func TestConvertOTelHistograms(t *testing.T) {
 		require.Equal(t, 0, dropped)
 		if convertHistogramsToNHCB {
 			ts := make([]mimirpb.PreallocTimeseries, 0, len(mimirTS))
+			// Filter out target_info series
 			for i := range mimirTS {
 				var metricName string
 				for _, lbl := range mimirTS[i].Labels {
@@ -386,7 +387,9 @@ func TestConvertOTelHistograms(t *testing.T) {
 			require.Len(t, ts[0].Histograms, 1)
 		} else {
 			require.Len(t, mimirTS, 6)
-			require.Len(t, mimirTS[0].Histograms, 0)
+			for i := range mimirTS {
+				require.Len(t, mimirTS[i].Histograms, 0)
+			}
 		}
 	}
 }

